server/dashboard/controller.go

```diff
@@ -243,6 +247,31 @@ func dashboardResultHandler(r *http.Request, w http.ResponseWriter, client *mong
 		filter, findOptions := buildReturnQuery(query)
 		// Sort by `timestamp` field descending.
 		findOptions.SetSort(bson.D{{Key: "timestamp", Value: -1}})
+
+		itemsPerPage, _ := strconv.ParseInt(query.Get("limit"), 10, 64)
+		if itemsPerPage == 0 {
+			itemsPerPage = 50
+		}
+
+		// Needed for pagination.
+		totalLogs, err := collection.CountDocuments(ctx, filter)
```

Comment:
why do we need to count here the total logs?

is it not possible to count them using 
```go
logs, err := db.GetLogsWithFilter(ctx, client, collection, filter, findOptions)
...
len(logs)
```
after line 275?

Reply:
1. why do we need to count here the total logs? so we can show how many page we need to show in pagination
2. nope, It's not possible because the result has been limited

Reply:
yeah sorry, I was not taking into account the fact that we need to show how many pages there are..